cli/deploy-profiling-job.sh

```diff
@@ -0,0 +1,123 @@
+else 
+  echo "Compute $COMPUTE_NAME creation failed"
+  exit 1
+fi
+
+# create role assignment for acessing workspace resources
+access_token=`az account get-access-token --query accessToken -o tsv`
+compute_info=`curl https://management.azure.com/subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE_NAME/computes/$COMPUTE_NAME?api-version=2021-03-01-preview -H "Content-Type: application/json" -H "Authorization: Bearer $access_token"`
+if [[ $? -ne 0 ]]; then echo "Failed to get info for compute $COMPUTE_NAME" && exit 1; fi
+identity_object_id=`echo $compute_info | jq '.identity.principalId' | sed "s/\"//g"`
+az role assignment create --role Contributor --assignee-object-id $identity_object_id --scope /subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE_NAME
+if [[ $? -ne 0 ]]; then echo "Failed to create role assignment for compute $COMPUTE_NAME" && exit 1; fi
+# </create_compute_cluster_for_hosting_the_profiler>
+
+# <create_endpoint>
+echo "Creating Endpoint $ENDPOINT_NAME ..."
+az ml endpoint create --name $ENDPOINT_NAME -f endpoints/online/managed/simple-flow/1-create-endpoint-with-blue.yml
+# </create_endpoint>
+
+# <check_endpoint_Status>
+endpoint_status=`az ml endpoint show --name $ENDPOINT_NAME --query "provisioning_state" -o tsv`
+echo $endpoint_status
+if [[ $endpoint_status == "Succeeded" ]]; then
+  echo "Endpoint $ENDPOINT_NAME created successfully"
+else 
+  echo "Endpoint $ENDPOINT_NAME creation failed"
+  exit 1
+fi
+
+deploy_status=`az ml endpoint show --name $ENDPOINT_NAME --query "deployments[?name=='$DEPLOYMENT_NAME'].provisioning_state" -o tsv`
+echo $deploy_status
+if [[ $deploy_status == "Succeeded" ]]; then
+  echo "Deployment $DEPLOYMENT_NAME completed successfully"
+else
+  echo "Deployment $DEPLOYMENT_NAME failed"
+  exit 1
+fi
+# </check_endpoint_Status>
+
+# <create_profiling_job_yaml_file>
+# please specify environment variable "IDENTITY_ACCESS_TOKEN" when working with ml compute with no appropriate MSI attached
+sed \
+  -e "s/<% ENDPOINT_NAME %>/$ENDPOINT_NAME/g" \
+  -e "s/<% DEPLOYMENT_NAME %>/$DEPLOYMENT_NAME/g" \
+  -e "s/<% PROFILING_TOOL %>/$PROFILING_TOOL/g" \
+  -e "s/<% DURATION %>/$DURATION/g" \
+  -e "s/<% CONNECTIONS %>/$CONNECTIONS/g" \
+  -e "s/<% TARGET_RPS %>/$TARGET_RPS/g" \
+  -e "s/<% CLIENTS %>/$CLIENTS/g" \
+  -e "s/<% TIMEOUT %>/$TIMEOUT/g" \
+  -e "s/<% THREAD %>/$THREAD/g" \
+  -e "s/<% COMPUTE_NAME %>/$COMPUTE_NAME/g" \
+  endpoints/online/profiling/profiling_job_tmpl.yml > profiling_job.yml
+# </create_profiling_job_yaml_file>
+
+# <create_profiling_job>
+run_id=$(az ml job create -f profiling_job.yml --query name -o tsv)
+# </create_profiling_job>
+
+# <check_job_status_in_studio>
+az ml job show -n $run_id --web
+# </check_job_status_in_studio>
+
+# <stream_job_logs_to_console>
+az ml job stream -n $run_id
+sleep 10
```

Comment:
why sleep 10?

Reply:
for stream command to end properly, otherwise all following commands won't execute as expected